asdf/tagged.py

```diff
@@ -109,6 +118,8 @@ def tag_object(tag, instance):
         return TaggedDict(instance, tag)
     elif isinstance(instance, list):
         return TaggedList(instance, tag)
+    elif isinstance(instance, time.Time):
+        return TaggedTime(instance, tag)
```

Comment:
This is getting to the point where it really ought to be replaced with a dispatch table of some kind--probably as a public interface too so that additional types can be registered.

Since this code was already here though, I'll open a separate issue for that.